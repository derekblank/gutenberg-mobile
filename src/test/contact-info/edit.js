/**
 * External dependencies
 */
import {
	getEditorHtml,
	initializeEditor,
	addBlock,
	getBlock,
	setupCoreBlocks,
	screen,
} from 'test/helpers';

/**
 * Internal dependencies
 */
import {
	registerJetpackBlocks,
	setupJetpackEditor,
} from '../../jetpack-editor-setup';

setupCoreBlocks();

beforeAll( () => {
	// Register Jetpack blocks
	setupJetpackEditor( {
		blogId: 1,
		isJetpackActive: true,
	} );
	registerJetpackBlocks( {
		capabilities: {
			contactInfoBlock: true,
		},
	} );
} );

describe( 'Contact Info block', () => {
	it( 'should successfully insert the block into the editor', async () => {
		await initializeEditor();

		// Add block
		await addBlock( screen, 'Contact Info' );

		// Get block
		const block = await getBlock( screen, 'Contact Info' );

		expect( block ).toBeVisible();
		expect( getEditorHtml() ).toMatchSnapshot();
	} );
} );
