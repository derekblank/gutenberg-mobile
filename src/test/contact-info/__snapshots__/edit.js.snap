// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Info block should successfully insert the block into the editor 1`] = `
"<!-- wp:jetpack/contact-info -->
<div class="wp-block-jetpack-contact-info"><!-- wp:jetpack/email /-->

<!-- wp:jetpack/phone /-->

<!-- wp:jetpack/address /--></div>
<!-- /wp:jetpack/contact-info -->"
`;
